meilisearch/client.py

```diff
@@ -80,6 +81,34 @@ def get_index(self, uid):
         """
         return Index.get_index(self.config, uid=uid)
 
+    def get_or_create_index(self, uid, options=None):
+        """Get an index, or create it if it doesn't exist.
+
+        Parameters
+        ----------
+        uid: str
+            UID of the index
+        options: dict, optional
+            Options passed during index creation (ex: primaryKey)
+
+        Returns
+        -------
+        index : Index
+            an instance of Index containing the information of the retrieved or newly created index
+        Raises
+        ------
+        HTTPError
+            In case of any other error found here https://docs.meilisearch.com/references/#errors-status-code
+        """
+        try:
+            index = Index(self.config, uid)
+            index.create(self.config, uid, options)
+        except MeiliSearchApiError as err:
+            if err.message != 'index already exists':
+                raise err
+            index = self.get_index(uid)
```

Comment:
No need to call `get_index` again, you already created an `Index` instance in `index = Index(self.config, uid)`

My suggestion:
```python
index = Index(self.config, uid) # This line will never trigger a MeiliSearchApiError
try:
    index.create(self.config, uid, options)
except MeiliSearchApiError as err:
    if err.code != 'index_already_exists': # pretty sure this line involves changes in MeiliSearchApiError ^^
        raise err
return index
```

Reply:
Thanks to both, I have updated the code!